include/alpaka/kernel/TaskKernelGenericSycl.hpp

```diff
@@ -276,6 +278,34 @@ namespace alpaka::trait
     {
         using type = TIdx;
     };
+
+    //! \brief Specialisation of the class template FunctionAttributes
+    //! \tparam TDim The dimensionality of the accelerator device properties.
+    //! \tparam TIdx The idx type of the accelerator device properties.
+    //! \tparam TKernelFn Kernel function object type.
+    //! \tparam TArgs Kernel function object argument types as a parameter pack.
+    template<typename TDim, typename TIdx, typename TKernelFn, typename... TArgs>
+    struct FunctionAttributes<AccGenericSycl<TDim, TIdx>, KernelBundle<TKernelFn, TArgs...>>
+    {
+        //! \param kernelBundle Kernel bundeled with it's arguments. The function attributes of this kernel will be
+        //! determined. Max threads per block is one of the attributes.
+        //! \return KernelFunctionAttributes instance. The default version always returns an instance with zero
+        //! fields. For CPU, the field of max threads allowed by kernel function for the block is 1.
+        ALPAKA_NO_HOST_ACC_WARNING
+        ALPAKA_FN_HOST_ACC static auto getFunctionAttributes(
+            [[maybe_unused]] KernelBundle<TKernelFn, TArgs...> const& kernelBundle) -> alpaka::KernelFunctionAttributes
+        {
+            alpaka::KernelFunctionAttributes kernelFunctionAttributes;
+            using Acc = AccGenericSycl<TDim, TIdx>;
+            auto const platformAcc = alpaka::Platform<Acc>{};
+            auto const dev = alpaka::getDevByIdx(platformAcc, 0);
```

Comment:
This is wrong: the different SYCL devices may have different properties, and so return different properties for the kernel.

Reply:
Ok you mean there could certainly be more devices than 1. I see.